superset-frontend/src/dashboard/components/gridComponents/Tabs.test.tsx

```diff
@@ -0,0 +1,235 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import { render, screen, waitFor } from 'spec/helpers/testing-library';
+import DashboardComponent from 'src/dashboard/containers/DashboardComponent';
+import DragDroppable from 'src/dashboard/components/dnd/DragDroppable';
+import DeleteComponentButton from 'src/dashboard/components/DeleteComponentButton';
+import getLeafComponentIdFromPath from 'src/dashboard/util/getLeafComponentIdFromPath';
+
+import Tabs from './Tabs';
+
+jest.mock('src/dashboard/containers/DashboardComponent', () =>
+  jest.fn(props => (
+    <button
+      type="button"
+      onClick={() =>
+        props.onDropOnTab({ destination: { id: 'TAB-YT6eNksV-' } })
+      }
+      data-test="DashboardComponent"
+    >
+      DashboardComponent
+    </button>
+  )),
+);
+
+jest.mock('src/dashboard/components/DeleteComponentButton', () =>
+  jest.fn(props => (
+    <button
+      type="button"
+      data-test="DeleteComponentButton"
+      onClick={props.onDelete}
+    >
+      DeleteComponentButton
+    </button>
+  )),
+);
+jest.mock('src/dashboard/util/getLeafComponentIdFromPath', () => jest.fn());
+
+jest.mock('src/dashboard/components/dnd/DragDroppable', () =>
+  jest.fn(props => {
+    const childProps = props.editMode
+      ? {
+          dragSourceRef: props.dragSourceRef,
+          dropIndicatorProps: props.dropIndicatorProps,
+        }
+      : {};
+    return (
+      <div>
+        <button type="button" data-test="DragDroppable" onClick={props.onDrop}>
+          DragDroppable
+        </button>
+        {props.children(childProps)}
+      </div>
+    );
+  }),
+);
+
+const creteProps = () => ({
```

Comment:
spelling nit: createProps
Also why does this need to be a function. There are not paramerters being passed in or reassignments. Does it make sense to make this an object.

Reply:
Fixed!

We talked about this in another PR: https://github.com/apache/superset/pull/13753#discussion_r600862661